include/deal.II/matrix_free/dof_info.templates.h

```diff
@@ -1262,39 +1266,102 @@ namespace internal
                         if (touched_by[myindex] ==
                             numbers::invalid_unsigned_int)
                           touched_by[myindex] = chunk;
+                        touched_last[myindex] = chunk;
                       }
                   }
           }
+
       // ensure that all indices are touched at least during the last round
       for (auto &index : touched_by)
         if (index == numbers::invalid_unsigned_int)
-          index = task_info.cell_partition_data.back() - 1;
+          index =
+            task_info
+              .partition_row_index[task_info.partition_row_index.size() - 2] -
```

Comment:
The `-2` is because you are interested in the first index?

Reply:
The last index is associated to cells corresponding to faces that are not worked on locally, which are not part of cell/face loops but needed in other instances.